util/cpuprofile/cpuprofile.go

```diff
@@ -155,43 +157,46 @@ func (p *parallelCPUProfiler) profilingLoop() {
 	checkTicker := time.NewTicker(DefProfileDuration)
 	defer func() {
 		checkTicker.Stop()
+		pprof.StopCPUProfile()
 		p.wg.Done()
 	}()
 	for {
 		select {
 		case <-p.ctx.Done():
 			return
 		case <-p.notifyRegister:
+			// If already in profiling, don't do anything.
+			if p.profileData != nil {
+				continue
+			}
 		case <-checkTicker.C:
 		}
 		p.doProfiling()
 	}
 }
 
 func (p *parallelCPUProfiler) doProfiling() {
+	if p.profileData != nil {
+		pprof.StopCPUProfile()
```

Comment:
This implementation is buggy. Suppose that someone has already started a profiling before starting the parallel cpu profiler:

At L193 you will have err != nil since the global profiler is already started by someone else.
Then at L179 you stops the other people's cpu profiler unexpectedly.

Reply:
No, since in `sendToConsumers`, `p.profileData` will be set to `nil` after send.